engine/src/main/java/org/terasology/rendering/dag/nodes/AmbientOcclusionNode.java

```diff
@@ -150,6 +148,8 @@ public void process() {
         ssaoMaterial.setFloat2("texelSize", 1.0f / outputFboWidth, 1.0f / outputFboHeight, true);
         ssaoMaterial.setFloat2("noiseTexelSize", NOISE_TEXEL_SIZE, NOISE_TEXEL_SIZE, true);
 
+        ssaoMaterial.setFloat3("ssaoSamples", ssaoSamples);
+
```

Comment:
This change might (should) confuse you.

Simple explanation : after task list regeneration, this property was not persisting for some reason, leading to black screens. Shifting this to process fixes everything.